tests/tracking/test_rest_tracking.py

```diff
@@ -456,3 +456,20 @@ def test_search_pagination(mlflow_client, backend_store_uri):
     result = mlflow_client.search_runs([experiment_id], max_results=4, page_token=result.token)
     assert [r.info.run_id for r in result] == runs[8:]
     assert result.token is None
+
+
+def test_get_experiment_by_name(mlflow_client, backend_store_uri):
+    name = 'test_get_experiment_by_name'
+    experiment_id = mlflow_client.create_experiment(name)
+    res = mlflow_client.get_experiment_by_name(name)
```

Comment:
Should we also check that the client did not have to fallback on list experiments?

Reply:
Yep good call, added a test for this. We also actually verify this in https://github.com/mlflow/mlflow/pull/1775/files#diff-339110c4cf69239298fe72c02f0d2cc6R279 (we verify that we only needed a single GetExperimentByName API request to get the experiment), although that test mocks out HTTP responses, so having the e2e test is definitely better.